hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/statemachine/commandhandler/DeleteBlocksCommandHandler.java

```diff
@@ -645,12 +645,16 @@ private void updateMetaData(KeyValueContainerData containerData,
 
       // update pending deletion blocks count and delete transaction ID in
       // in-memory container status
-      long pendingBytes = containerData.getBlockPendingDeletionBytes() + delTX.getTotalBlockSize();
+      long pendingBytes = containerData.getBlockPendingDeletionBytes();
+      if (delTX.hasTotalBlockSize()) {
```

Comment:
It's better to include all pendingBytes operations under the if scope, like this

// update in-memory pending deletion blocks size container status and persistent
If (VersionedDatanodeFeatures.isFinalized(HDDSLayoutFeature.DATA_DISTRIBUTION)) && delTX.hasTotalBlockSize()) {
      long pendingBytes = containerData.getBlockPendingDeletionBytes();
      pendingBytes += delTX.getTotalBlockSize();
      metadataTable
          .putWithBatch(batchOperation,
              containerData.getPendingDeleteBlockBytesKey(),
              pendingBytes);
 }

The base rule is if the feature is not finalized, we should avoid persist any data introduced by the feature permanently.